### PR TITLE
Sth/plugin proto

### PR DESCRIPTION
**Pull request by Sean-Holcomb**
## What does this PR change?
This PR changes the way that Custom Costs are recieved

## Does this PR relate to any other PRs?
* 

## How will this PR impact users?
* 

## Does this PR address any GitHub or Zendesk issues?
* Closes ...

## How was this PR tested?
* 

## Does this PR require changes to documentation?
* 

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
*

### CODE REVIEW

File: core/pkg/model/pb/messages.pb.go [hunk 1]
```diff
@@ -0,0 +1,876 @@
...
+)
+
+const (
+	// Verify that this generated code is sufficiently up-to-date.
+	_ = protoimpl.EnforceVersion(20 - protoimpl.MinVersion)
+	// Verify that runtime/protoimpl is sufficiently up-to-date.
+	_ = protoimpl.EnforceVersion(protoimpl.MaxVersion - 20)
+)
+
+type CustomCostResponse struct {
```
Comment (michaelmdresser) [inline core/pkg/model/pb/messages.pb.go:24]: If we're going to go to the effort of defining a nice proto schema for this data, why not just eliminate the other `CustomCostResponse` type entirely? Or is that an antipattern?

File: generate.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/zsh
```
Comment (michaelmdresser) [inline generate.sh]: For portability, I think `#!/usr/bin/env sh` or `#!/usr/bin/env bash` is better

File: core/pkg/model/pb/messages.pb.go (hunk 1 above)
Comment (ameijer) [reply core/pkg/model/pb/messages.pb.go:24]: Sean and I discussed this very thing. I was originally against it, but maybe that is worth revisiting

File: generate.sh (hunk 2 above)
Comment (Sean-Holcomb) [reply generate.sh]: yeah sorry, I was just trying to make it work quickly on my machine. I will swap in one of those

File: core/pkg/model/pb/messages.pb.go (hunk 1 above)
Comment (Sean-Holcomb) [reply core/pkg/model/pb/messages.pb.go:24]: That is my position for the most part, There may be some value to having the other definition for data manipulation in opencost, but we could also just extend the proto struct with any helper functions we want. For kubecost, I want the transition to duck to come from the plugin as protobuf directly and go directly to duck schema and into the db

**Review by ameijer — APPROVED** [review]

**Pull request merged**